src/GraphUtils.ts

```diff
@@ -330,14 +333,18 @@ export function findInterceptX(canvasProperties: CanvasProperties, pts: Point[])
 
     const intercepts = [];
 
-    if (pts[0].y == canvasProperties.centerPx.y) intercepts.push(pts[0]);
-    for (let i = 1; i < pts.length; i++) {
-        if (pts[i].y == canvasProperties.centerPx.y) {
-            intercepts.push(new Point(pts[i].x, pts[i].y));
+    for (let i = 0; i < pts.length-1; i++) {
+        if (pts[i].y === canvasProperties.centerPx.y) {
+            const start = i;
+            for (; i < pts.length && pts[i].y === canvasProperties.centerPx.y; i++);
```

Comment:
Incrementing i within the for loop would be more readable, but this approach may be more efficient. Up to you.

Reply:
I imagine it will compile the same, so we'll go for readability